src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -13,8 +13,8 @@ partial class Binder
     {
         private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            var expression = BindExpression(node.Expression, diagnostics);
-            var hasErrors = node.HasErrors || IsOperandErrors(node, expression, diagnostics);
+            var expression = BindValue(node.Expression, diagnostics, BindValueKind.RValue);
+            var hasErrors = node.HasErrors || IsOperandErrors(node, ref expression, diagnostics) || expression.HasErrors;
```

Comment:
Is `|| expression.HasErrors` necessary? `IsOperandErrors()` returns `expression.HasAnyErrors`.